class-wc-calypso-bridge-dotcom-features.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Functions for determining dotCom features in Calypso Bridge.
+ *
+ * @package WC_Calypso_Bridge/Classes
+ * @since   2.0.0
+ * @version 2.0.0
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+if ( ! function_exists( 'wc_calypso_bridge_is_ecommerce_plan' ) ) {
+	/**
+	 * Returns if a site is an eCommerce plan site or not.
+	 *
+	 * @return bool True if the site is an ecommerce site.
+	 */
+	function wc_calypso_bridge_is_ecommerce_plan() {
+		return WC_Calypso_Bridge_DotCom_Features::is_ecommerce_plan();
+	}
+}
+
+if ( ! function_exists( 'wc_calypso_bridge_is_ecommerce_trial_plan' ) ) {
+	/**
+	 * Returns if a site is an eCommerce trial plan site or not.
+	 *
+	 * @return bool True if the site is an ecommerce trial site.
+	 */
+	function wc_calypso_bridge_is_ecommerce_trial_plan() {
+		return WC_Calypso_Bridge_DotCom_Features::is_ecommerce_trial_plan();
+	}
+}
+
+if ( ! function_exists( 'wc_calypso_bridge_is_business_plan' ) ) {
+
+	/**
+	 * Returns if a site is a Business plan site or not.
+	 *
+	 * @return bool True if the site is a business site.
+	 */
+	function wc_calypso_bridge_is_business_plan() {
+		return WC_Calypso_Bridge_DotCom_Features::is_business_plan();
+	}
+}
+
+/**
+ * WC Calypso Bridge DotCom Features class.
+ */
+class WC_Calypso_Bridge_DotCom_Features {
+
+	/**
+	 * Is Ecommerce plan.
+	 *
+	 * @var bool
+	 */
+	protected static $is_ecommerce_plan = null;
+
+	/**
+	 * Is Business plan.
+	 *
+	 * @var bool
+	 */
+	protected static $is_business_plan = null;
+
+	/**
+	 * Is Ecommerce Trial plan.
+	 *
+	 * @var bool
+	 */
+	protected static $is_ecommerce_trial_plan = null;
+
+	/**
+	 * Determine if site is Ecommerce and cache it.
+	 *
+	 * @var bool
+	 */
+	public static function is_ecommerce_plan() {
+		if ( is_null( self::$is_ecommerce_plan ) ) {
+			self::$is_ecommerce_plan = (bool) function_exists( 'wpcom_site_has_feature' ) ? wpcom_site_has_feature( \WPCOM_Features::ECOMMERCE_MANAGED_PLUGINS ) : false;
```

Comment:
A few thoughts here:
1. Both the eCommerce trial and the main eCommerce plan have this feature, so either the logic or the naming needs to be changed.
2. For the case where the function doesn't exist yet, we have no way of knowing that the value was set before `wpcom_site_has_feature()` was defined, and the value should be checked. (This is an extreme edge case, but I want to flag it -- there may well be cases where `wpcom_site_has_feature()` is never defined, so we do need to pick our battles.)
3. This is a nit, but there is a _lot_ of logic on this one line! Pushing this to a multiline condition would make it easier for me to read and understand.
```suggestion
			self::$is_ecommerce_plan = (bool) function_exists( 'wpcom_site_has_feature' )
				? wpcom_site_has_feature( \WPCOM_Features::ECOMMERCE_MANAGED_PLUGINS )
				: false;
```